### ISSUE TITLE
Dynamic function name not recognized and causes errors

### ISSUE THREAD

**Issue by rgtk** [palantir/tslint#4255]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class A {
  [util.inspect.custom] {
    return '<A>';
  }
}
```

with `tslint.json` not provided (using default).

#### Actual behavior
```
ERROR: 2:24  semicolon   Missing semicolon
ERROR: 3:1   ter-indent  Expected indentation of 2 spaces but found 4.
ERROR: 4:1   ter-indent  Expected indentation of 0 spaces but found 2.
```
#### Expected behavior
No errors.

**Comment by JoshuaKGoldberg** [comment]
@rgtk that isn't valid TypeScript unfortunately. You're missing either a `()` or a `=` after the `]`.

See #3808 for discussion on what TSLint should do in the case of syntax errors.
